pubsublite/psl.go

```diff
@@ -0,0 +1,32 @@
+// Package pubsublite is an API for pubsub providers that guarantee
+// at-most-once delivery, and can dynamically publish to topics with names that
+// can be chosen at runtime, and can add or remove any number of subscriptions
+// dynamically. These providers do not have a notion of message
+// acknowledgement.
+package pubsublite
+
```

Comment:
I think we should do one of two things:
1. If we want to think of this as a kind of pubsub, then to minimize confusion we should have the same API as our `pubsub` package, minus `Ack`. 
2. Maybe despite everyone calling it pubsub, it's really something else—message broadcasting—and we should have a new, top-level package with different names for everything.

Reply:
I'm in favor of option 2, calling it maybe `broadcast` or `messaging`.

Reply:
I don't think #1 makes sense because we can no longer rely on the existence of subscriptions defined outside the API that queue up messages. The API needs to specifically do something to help prevent dropping messages on the floor, either buffering or spawning goroutines that handle the messages.

Reply:
Depending on usage you may want to back up and reconsider the pubsub package itself. You probably need to abstract out QoS to some degree. The only reason for two packages is because of incorrect assumptions in pubsub.

Reply:
About thing 2, I still think it's fine to talk about publishing and subscribing with a different quality of service (QoS) level: at-most-once. That's the standard nomenclature as far as I know. That said, I do like the idea of calling the new subpackage `broadcast` since it's a whole, reasonably short English word that suggests the transience of messaging in these providers.

Reply:
I disagree. The routing is the same in most packages just QoS changes. For instance NATS is pub/sub, req/reply and distributed queues. Broadcast is not a good name for that in my opinion.